### PR TITLE
Add code comments and examples to `Map` inline example in API docs

### PR DESCRIPTION
**Pull request by domlet**
- adds code comments
- adds 3 related examples

Note: the related examples should not appear at the bottom of the page – ideally they should appear after `examples` and before `instance members`. I will work on a `mapbox-gl-js-docs` PR to implement that change.

Tagging @HeyStenson @ryanhamley for review.

## Launch Checklist
 - [x] briefly describe the changes in this PR
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

File: src/ui/map.js [hunk 1]
```diff
@@ -268,11 +268,12 @@ const defaultOptions = {
...
- *   container: 'map',
- *   center: [-122.420679, 37.772537],
- *   zoom: 13,
- *   style: style_object,
- *   hash: true,
+ *   container: 'map', // container ID
+ *   center: [-122.420679, 37.772537], // starting position [lng, lat]
+ *   zoom: 13, // starting zoom
+ *   style: 'mapbox://styles/mapbox/streets-v11', // style URL or style object
+ *   hash: true, // sync options to URL
```
Comment (ryanhamley) [inline src/ui/map.js:275]: the hash only contains the center, zoom, pitch and bearing. `options` is a much broader set of properties

**Review by ryanhamley — APPROVED** [review]
I had a nit about the hash comment but otherwise looks good

**Review by HeyStenson — APPROVED** [review]
🍊 🌞

File: src/ui/map.js (hunk 1 above)
Comment (domlet) [inline src/ui/map.js:275]:
```suggestion
 *   hash: true, // Sync `center`, `zoom`, `pitch`, and `bearing` with URL
```

Comment (domlet) [reply src/ui/map.js]: Thank you – it's a helpful nit. Updated the code comment to be more specific. 👍🏼

File: src/ui/map.js [hunk 2]
```diff
@@ -268,11 +268,12 @@ const defaultOptions = {
...
- *   container: 'map',
- *   center: [-122.420679, 37.772537],
- *   zoom: 13,
- *   style: style_object,
- *   hash: true,
+ *   container: 'map', // container ID
+ *   center: [-122.420679, 37.772537], // starting position [lng, lat]
+ *   zoom: 13, // starting zoom
+ *   style: 'mapbox://styles/mapbox/streets-v11', // style URL or style object
+ *   hash: true, // Sync `center`, `zoom`, `pitch`, and `bearing` with URL
```
Comment (domlet) [inline src/ui/map.js:275]:
```suggestion
 *   hash: true, // sync `center`, `zoom`, `pitch`, and `bearing` with URL
```

**Review by ryanhamley — APPROVED** [review]

**Pull request merged**